### PR TITLE
Minor changes and Health Check

### PR DESCRIPTION
**Pull request by xbenyx**
Summary: In this update, some functions have been refactored and moved to utility functions. Additionally, fixes have been applied to address identified issues. Also Health checks is adapted to Angular Material.

### CODE REVIEW

File: src/app/config/health-checks/new-health-check/new-health-checks.component.ts [hunk 1]
```diff
@@ -1,61 +1,152 @@
...
   selector: 'app-new-health-checks',
   templateUrl: './new-health-checks.component.html'
 })
-@PageTitle(['New Health Check'])
-export class NewHealthChecksComponent implements OnInit {
-
-  // Form create Health Check
+export class NewHealthChecksComponent implements OnInit, OnDestroy {
+  /** Form group for Health Checks */
   createForm: FormGroup;
```
Comment (itbf9) [inline src/app/config/health-checks/new-health-check/new-health-checks.component.ts:24]: I don't really like the prefix `create` on a variable name. It suggests that it is an action to me. How about just `form` or `healthChecksForm`?

File: src/app/config/health-checks/new-health-check/new-health-checks.component.ts [hunk 2]
```diff
@@ -1,61 +1,152 @@
...
-      'crackerBinaryId': new FormControl('', [Validators.required])
+      checkType: new FormControl(0),
+      hashtypeId: new FormControl(null || 0, [Validators.required]),
+      crackerBinaryId: new FormControl('', [Validators.required]),
+      crackerBinaryType: new FormControl('')
     });
 
+    this.createForm
+      .get('crackerBinaryType')
+      .valueChanges.subscribe((newvalue) => {
```
Comment (itbf9) [inline src/app/config/health-checks/new-health-check/new-health-checks.component.ts:87]: This subscription is never unsubscribed

File: src/app/files/new-files/new-files.component.ts [hunk 3]
```diff
@@ -206,16 +205,10 @@ export class NewFilesComponent implements OnInit, OnDestroy {
...
-    if (form.sourceType === 'inline') {
-      fname = form.filename;
-      sourcadata = Buffer.from(form.sourceData).toString('base64');
-    } else {
-      sourcadata = this.fileName;
-      fname = this.fileName;
-    }
+    const sourceType = form.sourceType || 'import';
+    const isInline = sourceType === 'inline';
+    const fname = isInline ? form.filename : this.fileName;
```
Comment (itbf9) [inline src/app/files/new-files/new-files.component.ts:210]: Why is filename camelCased sometimes? :)

File: src/app/shared/utils/forms.ts [hunk 4]
```diff
@@ -140,3 +140,35 @@ export function compareVersions(a, b): number {
...
+    : originalPath;
+}
+
+/**
+ * Handles the encoding of source data based on the selected file type.
+ *
+ * @param {string} fileSource - The source data to be encoded.
+ * @returns {string} The encoded source data in base64 format.
+ */
+import { Buffer } from 'buffer';
```
Comment (itbf9) [inline src/app/shared/utils/forms.ts:171]: Why is this import down here?

**Review by itbf9 — CHANGES_REQUESTED** [review]
A few minor remarks but overall lgtm

File: src/app/config/health-checks/new-health-check/new-health-checks.component.ts (hunk 1 above)
Comment (xbenyx) [reply src/app/config/health-checks/new-health-check/new-health-checks.component.ts:24]: I have been using form in other components, makes more sense. I'll change it.

File: src/app/config/health-checks/new-health-check/new-health-checks.component.ts (hunk 2 above)
Comment (xbenyx) [reply src/app/config/health-checks/new-health-check/new-health-checks.component.ts]: Added.

File: src/app/files/new-files/new-files.component.ts (hunk 3 above)
Comment (xbenyx) [reply src/app/files/new-files/new-files.component.ts:210]: Thanks. Changes to camelCased

File: src/app/shared/utils/forms.ts (hunk 4 above)
Comment (xbenyx) [reply src/app/shared/utils/forms.ts:171]: I'm using handleEncode in New files and New hashlists, so probably I could move the import to the top of utils? or use dependency injection but I'll need to import buffer in both components. What do you think?

Comment (itbf9) [reply src/app/shared/utils/forms.ts:171]: I just meant it was a little misplaced in the middle of the file, just move it to the top i would say

**Review by itbf9 — APPROVED** [review]

**Pull request merged**